subsys/net/ip/tcp2_priv.h

```diff
@@ -35,26 +37,55 @@
 #if IS_ENABLED(CONFIG_NET_TEST_PROTOCOL)
 #define tcp_pkt_alloc(_conn, _len)					\
 ({									\
-	sa_family_t _family = net_context_get_family((_conn)->context);	\
-	struct net_pkt *_pkt = net_pkt_alloc_with_buffer((_conn)->iface,\
-							 (_len),	\
-							 _family,	\
-							 IPPROTO_TCP,	\
-							 K_NO_WAIT);	\
+	struct net_pkt *_pkt;						\
+									\
+	if ((_len) > 0) {						\
+		_pkt = net_pkt_alloc_with_buffer(			\
+			(_conn)->iface,					\
+			(_len),						\
+			net_context_get_family((_conn)->context),	\
+			IPPROTO_TCP,					\
+			TCP_PKT_ALLOC_TIMEOUT);				\
+	} else {							\
+		if (!(_conn)->unacked_data) {				\
```

Comment:
Shall we just alloc empty packet in tcp_conn_alloc() and drop working with (_conn)->unacked_data here?